cmd/thanos/query.go

```diff
@@ -119,6 +122,13 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 			fileSD = file.NewDiscovery(conf, logger)
 		}
 
+		queryMaxDuration := time.Duration(*queryTimeout)
+		storeReadMaxDuration := time.Duration(*storeReadTimeout)
+
+		if storeReadMaxDuration > queryMaxDuration {
```

Comment:
Let's remove that. `Context` logic already ensures that

Reply:
+1